### DISCUSSION TITLE
Complex optimization example

### DISCUSSION THREAD

**Discussion by dietmarwo** [oemof/tespy#345] [Show and tell]
Hi, I am the maintainer of a Python optimization library https://github.com/dietmarwo/fast-cma-es
and wrote a TESPy optimization tutorial https://github.com/dietmarwo/fast-cma-es/blob/master/tutorials/PowerPlant.adoc which shows how to utilize your many core CPU to 
speed up optimization by parallelization. 
I would like to add a more complex optimization example involving multiple constraints and objectives. I am neither a TESPy nor a thermal engineering expert, but may be somone has an
interesting optimization problem and would like to speed up things, so we could cooperate.

**Comment by nkawerau** [comment]
Hi @dietmarwo,

I have created a Pyhton model using TESPy, with the goal of optimizing an alkaline water electrolyzer that generates large networks. At the moment I am trying to validate my model, but in the near future I would like to do optimization calculations. I'm pretty much a Python novice, and I asked a few months ago if there are possibilities for parallelized computations in TESPy. I would definitely be interested in speeding up the calculations and cooperating.

Thanks
Nicolas

**Reply by drdwo** [reply]
Hi Nicolas, please contact me at drdietmarwolz@yahoo.de when your model is testable. I think there is a good chance to get parallel optimization working for your model.

**Comment by fwitte** [comment]
Hi @dietmarwo,

thank you for reaching out in this channel :). In general, I would be very interested in this as well, as I will have something to contribute by the end of this year:

For my PhD I will work on some more complex case with multiple (and maybe some integer) decision variables. I need this to be finished on a quite tight schedule, therefore, I would prefore to extend the parallel optimization based on that work, so maybe that could be something we take into perspective as well. It will be some combination of thermodynamics and economics.

**Reply by dietmarwo** [reply]
There are some enhancements for integer decision variables in some of the fcmaes algorithms, so this would fit very nicely.
